rasa/engine/validation.py

```diff
@@ -82,6 +96,29 @@ def validate(
         )
 
 
+def _validate_cycle(schema: GraphSchema) -> None:
+    for target_name in schema.target_names:
```

Comment:
this would technically not catch cycles if there is an unused node in the graph 🤔 Wanted to avoid walking the graph to often (which would happen if we do this for every node)